test/components/prometheus-operator/prometheus_operator_test.go

```diff
@@ -67,3 +78,88 @@ func TestPrometheusOperatorDeployment(t *testing.T) {
 
 	testutil.WaitForDaemonSet(t, client, namespace, "prometheus-operator-prometheus-node-exporter", retryInterval, timeout)
 }
+
+// nolint:funlen
+func TestGrafanaLoadsEnvVars(t *testing.T) {
+	kubeconfig := testutil.KubeconfigPath(t)
+
+	t.Logf("using KUBECONFIG=%s", kubeconfig)
+
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		t.Fatalf("failed building rest client: %v", err)
+	}
+
+	client, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		t.Fatalf("failed creating new clientset: %v", err)
+	}
+
+	// We will wait until the Grafana Pods are up and running so we don't have to reimplement wait logic again.
+	testutil.WaitForDeployment(t, client, namespace, grafanaDeployment, retryInterval, timeout)
+
+	// Get grafana deployment object so that we can get the corresponding pod.
+	deploy, err := client.AppsV1().Deployments(namespace).Get(context.TODO(), grafanaDeployment, metav1.GetOptions{})
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			t.Fatalf("deployment %s not found", grafanaDeployment)
+		}
+
+		t.Fatalf("error looking up for deployment %s: %v", grafanaDeployment, err)
+	}
+
+	podList, err := client.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{
+		LabelSelector: metav1.FormatLabelSelector(deploy.Spec.Selector),
+	})
+	if err != nil {
+		t.Fatalf("could not list pods for the deployment %q: %v", grafanaDeployment, err)
+	}
+
+	if len(podList.Items) == 0 {
+		t.Fatalf("grafana pods not found")
+	}
+
+	// Exec into the pod.
```

Comment:
It'd be nice having this in a function that can run an arbitrary command in a pod by execing. However, I'm OK with having this here for now since no other test is using it.

Reply:
I also thought of putting it into a separate func so that others can use it. But I don't feel a need to do it now. Since none of the other tests need it.